### ISSUE TITLE
Counter Incorrect upon Export to PDF

### ISSUE THREAD

**Issue by LPalmerMSI** [asciidoctor/asciidoctor-pdf#1986]
Using a counter inside a table produces different results depending on the output.

**Product and Version:**
Asciidoctor IntelliJ Plugin Project
0.36.10

Creating a table with a counter like:

```
[options="header", cols="2"]
|===
a|{counter:swareq}
|Some Col 1

a|{counter:swareq}
|Some Col 2

```

This counter produces the expected output using preview in JetBrains Rider, with the first row having the value 1 in the first column.  However, when exporting this to PDF, the first column is starting at 10.  It's worth noting that I have another adoc in the same folder which also uses counters in this way.  When I initially created that file I did not see that behavior but now I am seeing it.  I don't know if this behavior is related to having more than one adoc in the same directory both using counters. 

In Preview:
![image](https://user-images.githubusercontent.com/72145572/147267812-68176e15-4868-43dd-ba9d-7924dc0f7481.png)

Exported to PDF:
![image](https://user-images.githubusercontent.com/72145572/147267887-ace6723e-7d82-4616-bedf-d8c0d59efa02.png)

**Comment by mojavelinux** [comment]
The `a` style is only needed when the cell contains block elements. In your case, the `a` style is not necessary, so that is the simple solution to problem you reported.

For when the `a` style is needed, Asciidoctor PDF 1.6.x has a known issue that it does not restore the counter after computing the height of the cell in the scratch document. This is fixed in Asciidoctor PDF 2. Therefore, I'm going to mark this as a confirmed duplicate and label it as fixed in 2.0.0.

Thank you for providing the test case.

**Comment by mojavelinux** [comment]
Here's the test that confirms the fix: https://github.com/asciidoctor/asciidoctor-pdf/blob/b4c6591c82526ae1c657a533c2eefdb904c9d1aa/spec/table_spec.rb#L1608-L1617
